--- DisconnectedRealmsScreen.java
+++ DisconnectedRealmsScreen.java
@@ -2,12 +2,12 @@
 
 import java.util.List;
 import net.minecraft.text.Text;
 
 public class DisconnectedRealmsScreen extends RealmsScreen {
-	private String title;
-	private Text reason;
+	private final String title;
+	private final Text reason;
 	private List<String> lines;
 	private final RealmsScreen parent;
 	private int textHeight;
 
 	public DisconnectedRealmsScreen(RealmsScreen realmsScreen, String string, Text text) {
